### PR TITLE
add qb package to Docker

### PR DESCRIPTION
**Pull request by trim21**
### Motivation for changes:

I'm using both qb and transmission…

### Detailed changes:
- add qbittorrent-api python package to docker image

### Addressed issues/feature requests:
- Fixes # .

### Config usage if relevant (new plugin or updated schema):
```
paste_config_here
```
### Log and/or tests output (preferably both):
```
paste output here
```
#### To Do:

- [ ] Stuff..

### CODE REVIEW

**Comment by trim21** [comment]
looks like dockerfile is not actually tested

**Comment by gazpachoking** [comment]
I'm good with this.
It made me start thinking though, should we add a dependency group to the pyproject for optional deps that are included in the docker image rather than putting them directly in the dockerfile? Would mean that poetry could resolve the specific versions that get installed and make sure they are compatible with our other deps. @paranoidi @X-dark @ksurl ?

**Comment by ksurl** [comment]
does that mean it will install at runtime? if it can pin versions rather than latest, it might be ok. unpredictable packages is against the philosophy of static images, but I'm sure other projects break this rule all the time if it works and is limited in scope.

**Comment by gazpachoking** [comment]
No, not proposing it installs at runtime. We could either have it generate a docker-requirements.txt similar to how we have other requirements files (which the dockerfile would then use) or we could switch the dockerfile to install flexget with poetry ratther than pip. In any case, the deps versions would still be resolved at image creation time.

**Comment by gazpachoking** [comment]
> looks like dockerfile is not actually tested

Yeah. Do you (or anyone) have a good idea of what we should test/how we should test it?

**Comment by trim21** [comment]
> > looks like dockerfile is not actually tested
> 
> Yeah. Do you (or anyone) have a good idea of what we should test/how we should test it?

build and run flexget --help

**Comment by ksurl** [comment]
> No, not proposing it installs at runtime. We could either have it generate a docker-requirements.txt similar to how we have other requirements files (which the dockerfile would then use) or we could switch the dockerfile to install flexget with poetry ratther than pip. In any case, the deps versions would still be resolved at image creation time.

Haha I actually made a local branch to pr a separate requirement file for docker but never pushed it.

**Comment by trim21** [comment]
I'll just let poetry to export a requiremenets-docker.txt to be requirements.txt with docker group

**Comment by trim21** [comment]
and why we install `nodejs` in docker image? which plugin need it?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -22,8 +22,8 @@ repos:
     - id: poetry-export
       args: ["--only", "dev", "-f", "requirements.txt", "-o", "dev-requirements.txt", "--without-hashes"]
     - id: poetry-export
-      args: ["--only", "plugin-test", "-f", "requirements.txt", "-o", "dev-requirements-extras.txt", "--without-hashes"]
```
Comment (gazpachoking) [inline .pre-commit-config.yaml:25]: Did you mean this one to come out?

Comment (trim21) [reply .pre-commit-config.yaml:25]: Sorry, I want to add a new one, didn't want to change existing

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -60,12 +60,11 @@ jobs:
       - name: Build Base Docker Image
         id: docker_build
         uses: docker/build-push-action@v4
-        if: github.event_name == 'push' && github.ref != 'refs/heads/master'
         with:
           context: ./
           file: ./Dockerfile
           builder: ${{ steps.buildx.output.name }}
-          push: true
+          push: ${{ github.event_name == 'push' && github.ref != 'refs/heads/master' }}
```
Comment (gazpachoking) [inline .github/workflows/docker-image.yml:67]: What's the motivation for this change? Does it change anything functionally?

Comment (ksurl) [reply .github/workflows/docker-image.yml:67]: Looks like it only pushes on tags rather than every commit?

Comment (gazpachoking) [reply .github/workflows/docker-image.yml:67]: It got changed from being in the 'if' block to down here. Should have put my comment on the range so that the diff showed up better. Just curious the motivation.
image

Comment (trim21) [reply .github/workflows/docker-image.yml:67]: it doesn't build docker image with previous `if` field， dockerfile is not tested in the pr

Comment (trim21) [reply .github/workflows/docker-image.yml:67]: it doesn't actually change anything， previous ci only push and build on the condition， now it build in any condition， and push in the condition

Comment (trim21) [reply .github/workflows/docker-image.yml:67]: https://github.com/Flexget/Flexget/actions/runs/4659863819/jobs/8247244212

Comment (gazpachoking) [reply .github/workflows/docker-image.yml:67]: Ahh, I get it now. Sounds good!
